pool/connection_pool.go

```diff
@@ -1109,8 +1126,54 @@ func (p *ConnectionPool) handlerDeactivated(conn *tarantool.Connection,
 	}
 }
 
-func (p *ConnectionPool) fillPools() bool {
+func (p *ConnectionPool) deactivateConnection(addr string,
+	conn *tarantool.Connection, role Role) {
+	p.deleteConnection(addr)
+	conn.Close()
+	p.handlerDeactivated(conn, role)
+}
+
+func (p *ConnectionPool) deactivateConnections() {
+	for address, endpoint := range p.addrs {
+		if endpoint == nil || endpoint.conn == nil {
+			continue
+		}
+		p.deactivateConnection(address, endpoint.conn, endpoint.role)
+	}
+}
+
+func (p *ConnectionPool) processConnection(conn *tarantool.Connection,
+	addr string, end *endpoint) bool {
+	role, err := p.getConnectionRole(conn)
+	if err != nil {
+		conn.Close()
+		log.Printf("tarantool: storing connection to %s failed: %s\n", addr, err)
+		return false
+	}
+
+	if !p.handlerDiscovered(conn, role) {
+		conn.Close()
+		return false
+	}
+	if p.addConnection(addr, conn, role) != nil {
+		conn.Close()
+		p.handlerDeactivated(conn, role)
+		return false
+	}
+
+	if conn.ConnectedNow() {
```

Comment:
I don't see a reason for the check. We already check that connection is alive in `getConnectionRole()` call.

Reply:
Updated.